keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -115,46 +115,68 @@ This proposal introduces the support of both persistent and ephemeral-style volu
-* Volumes are assumed to be originated from an inline pod spec.
+### Persistent inline mode
+For CSI drivers that indicate they support persistent inline volume definitions, the `CSIDriver.Lifecycle` value should be set to `Persistent` (default if not provided).  When a persistent inline driver is used to handle volume operations: 
+* Volume requests are assumed to have originated from an inline pod spec.
 * The `volumeHandle` value must be generated ahead of time and manually entered in the pod spec.
-* The driver will only receive CSI calls related to storage opearations for *attach*, *mount*, *unmount*, *detach*.
-* The driver will not receive CSI calls for *porovision* and *deletion* of volumes.
-
-#### Ephemeral inline mode
-Ephemeral inline volumes are more restricted than their persistent counterparts.  The `CSIDriver.lifecycle = ephemeral` must be set to indicate support for inline ephemeral volumes which will follow the lifecycle of its pod.  For drivers indicating support for ephemral mode: 
-* Volumes are assumed to be originated from an inline pod spec.
-* The `volumeHandle` is not required and is ignored if provided in the pod spec.
-* CSI will internally generate a `volumeHandle` value which is passed to the driver.
+* Omission of the volumeHandle will result in an error during the attachment phase which will fail.
+* The driver will only receive CSI calls related to storage operations for *attach*, *mount*, *unmount*, *detach*.
+* The driver will not receive CSI calls for *provision* and *deletion* of volumes.
+
+#### Persistent inline volume limitations
+During the attachment phase, information about pod and pod namespace is not available.  This introduces the following limitations for persistent inline volumes:
+* Persistent inline volume requests will not participate in attach/detach operations that require secrets.
+* Access to pod namespaces are not available during attachment.
+* This means secret references will not be resolved during attachment/detachment.
+* CSI operations during attachment/detachment will receive a nil secret references.
+
+Existing in-tree drivers only fully support attachment/detachment phases with PV/PVC-originated volumes. While adding attachment support for pod-originated is a nice feature, it will only work with the limitations outline above.
+
+#### Persistent inline volume referred by multiple pods 
+One or more pods can make references to persistent inline volumes with the same driver and volumeHandle.  When these pods are scheduled on the same node, the following will occur:
+* The internal CSI attacher code will search for existing attachment object with ID composed of `node name + driver name + volumeHandle`.
+* If the volume attachment does not exist yet, the attachment will be created for the volume on that node.
+* Subsequent pods that land on the node, with inline persistent volume requests with same driver and volumeHandle, will not trigger a new volume attachment object.
```

Comment:
call out the implication that the volumeattachment object used to attach the volume may not match the parameters specified by the pod (readonly and volumeAttributes)

Reply:
@liggitt I will call i tout for clarity.